masq/src/test_utils/mocks.rs

```diff
@@ -552,16 +557,28 @@ impl MasqTerminal for TerminalActiveMock {
     fn lock(&self) -> Box<dyn WriterLock + '_> {
         Box::new(self.in_memory_terminal.lock_writer_append().unwrap())
     }
+
+    fn lock_ultimately(
+        &self,
+        _streams: &mut StdStreams,
+        _stderr: bool,
+    ) -> Box<dyn WriterLock + '_> {
+        Box::new(self.in_memory_terminal.lock_writer_append().unwrap())
+    }
 }
 
 impl TerminalActiveMock {
     pub fn new() -> Self {
         Self {
-            in_memory_terminal: Interface::with_term("test only terminal", MemoryTerminal::new())
-                .unwrap(),
+            in_memory_terminal: Interface::with_term(
+                "test only terminal",
+                MemoryTerminal::new().clone(),
+            )
+            .unwrap(),
             user_input: Arc::new(Mutex::new(vec![])),
         }
     }
+
     pub fn read_line_result(self, line: String) -> Self {
```

Comment:
user_input should be called read_line_results, so that people accustomed to looking at standard-form mocks will recognize it.